components/locid/src/serde/locid.rs

```diff
@@ -0,0 +1,81 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::Locale;
+use alloc::string::ToString;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+impl Serialize for Locale {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for Locale {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct LocaleVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for LocaleVisitor {
+            type Value = Locale;
+
+            fn expecting(&self, formatter: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                write!(formatter, "a valid Unicode Locale Identifier")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                s.parse::<Locale>().map_err(serde::de::Error::custom)
+            }
+        }
+
+        deserializer.deserialize_string(LocaleVisitor)
+    }
+}
+
+#[test]
+fn serialize() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
`nit`: Can you name the tests in a bit more descriptive way? It helps people call them exclusively via `cargo test keyword`

Reply:
Of course. Since we are at it, shall I also rename the serialize/deserialize tests for `LanguageIdentifier`, which are named identically, in an additional commit or PR?

Reply:
yes please! Maybe name like `test_locid_serde_serialize` etc.

Reply:
I took the liberty of adding the change as an individual commit rather polluting the tracker with a trivial PR.